provider/k8s/helpers.go

```diff
@@ -338,3 +340,55 @@ func calculatePodCpuAndMem(m *metricsv1beta1.PodMetrics) (cpu float64, mem float
 	}
 	return toCpuCore(cpuTotal), toMemMB(memTotal)
 }
+
+func aggregateMetricByPeriod(m structs.Metric, period int64) structs.Metric {
```

Comment:
return `structs.Metric` as a reference

```suggestion
func aggregateMetricByPeriod(m structs.Metric, period int64) *structs.Metric {
```

Reply:
@nightfury1204 if you pass by value, the variable will be copied to another memory address location (using more memory and more processing). Passing by reference you will pass the memory address location, saving some memory space and processing, which is useful when passing a big slice or a big variable.

https://david-yappeter.medium.com/golang-pass-by-value-vs-pass-by-reference-e48aac8b2716

Reply:
:+1: 
addressed the changes